python/ray/tests/test_actor_failures.py

```diff
@@ -989,6 +989,45 @@ def verify():
     wait_for_condition(verify)
 
 
+def test_exit_actor_async_actor_nested_task(shutdown_only, tmp_path):
+    async_temp_file = tmp_path / "async_actor.log"
+    async_temp_file.touch()
+
+    @ray.remote
+    class AsyncActor:
+        def __init__(self):
+            def f():
+                print("atexit handler")
+                with open(async_temp_file, "w") as f:
+                    f.write("Async Actor\n")
```

Comment:
please use the `SignalActor` utility instead for cross-process synchronization in tests


lmk if you have questions on how to do it

Reply:
I copied this from other tests in this file. Should I also update the other tests (perhaps in follow-up PRs)?

Reply:
yes please do; a follow-up PR sounds great

Reply:
After offline discussion, there is a weird interaction between `atexit` handler. So I don't use `SignalActor` in this PR in the end.